c_src/ep_encoder.c

```diff
@@ -617,13 +617,9 @@ get_oneof_field(ErlNifEnv *env, ERL_NIF_TERM term, ep_node_t *node, ERL_NIF_TERM
         return NULL;
     }
 
-    if (arity != 2 || !enif_is_atom(env, array[0])) {
-        return NULL;
-    }
-
-    *out = array[1];
+    *out = term;
 
-    return bsearch(&(array[0]), node->fields, node->size, sizeof(ep_field_t), get_field_compare_name);
+    return bsearch((&(array[0])), node->fields, node->size, sizeof(ep_field_t), get_field_compare_sub_name);
```

Comment:
Don't we have to have the node->fields sorted by the new thing? Or is this true because we make sure the names and the types match

Reply:
I am basically trying out if this simpler change will work.
Yes, node-> fields must be sorted.. so now this will force us to maintain definitions with oneof fields to be in alphabetical order always. otherwise it will break!